src/main/scala/Castles.scala

```diff
@@ -21,11 +21,7 @@ object Castles extends OpaqueBitboard[Castles]:
     def blackQueenSide: Boolean = (c & A8.bitboard).nonEmpty
 
     def without(color: Color): Castles =
-      color match
-        case White =>
-          c & ~A1.bitboard & ~H1.bitboard
-        case Black =>
-          c & ~A8.bitboard & ~H8.bitboard
+      c & Bitboard.rank(color.lastRank)
```

Comment:
Shouldn't it be the opposite? If I'm reading this correctly, this is including only the color while it should be excluding?

Reply:
The `lastRank` of white is the 8th rank, aka the black `backRank`.

Reply:
Oh, but `&`, so you're right

Reply:
Ah right, yeah, then it's correct